extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetColumnLocation.java

```diff
@@ -370,10 +370,22 @@ private static <ATTR extends Any, RESULT> ToPage<ATTR, RESULT> makeToPage(
                 final PrimitiveType.PrimitiveTypeName typeName = type.getPrimitiveTypeName();
                 switch (typeName) {
                     case BOOLEAN:
-                        toPage = ToBooleanAsBytePage.create(pageType);
+                        if (pageType.equals(Boolean.class)) {
+                            toPage = ToBooleanAsBytePage.create(pageType);
+                        } else if (pageType.equals(short.class)) {
+                            toPage = ToShortPage.createFromBoolean(pageType);
+                        } else if (pageType.equals(int.class)) {
+                            toPage = ToIntPage.createFromBoolean(pageType);
+                        } else if (pageType.equals(long.class)) {
+                            toPage = ToLongPage.createFromBoolean(pageType);
+                        }
                         break;
                     case INT32:
-                        toPage = ToIntPage.create(pageType);
+                        if (pageType.equals(int.class)) {
+                            toPage = ToIntPage.create(pageType);
+                        } else if (pageType.equals(long.class)) {
```

Comment:
You've now set a precedent that we will check pageType here... but not done so for all types. Maybe we want to ensure there's no mismatch explicitly, and return a better error message?